include/geode/basic/progress_logger_manager.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2019 - 2022 Geode-solutions
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#pragma once
+
+#include <geode/basic/common.h>
+#include <geode/basic/pimpl.h>
+
+namespace geode
+{
+    class ProgressLoggerClient;
+} // namespace geode
+
+namespace geode
+{
+    class opengeode_basic_api ProgressLoggerManager
+    {
+    public:
+        ~ProgressLoggerManager();
+
+        static void register_client(
+            std::unique_ptr< ProgressLoggerClient >&& client );
+
+        static void start( const std::string& message, index_t nb_steps );
+
+        static void update( index_t current_step, index_t nb_steps );
+
+        static void complete();
+
+        static void failed();
```

Comment:
Maybe these could be protected by a key so only the ProgressLogger can call them.